pkg/kubelet/dockertools/manager.go

```diff
@@ -1586,3 +1595,34 @@ func (dm *DockerManager) SyncPod(pod *api.Pod, runningPod kubecontainer.Pod, pod
 
 	return nil
 }
+
+// verifyNonRoot returns an error if the container or image will run as the root user.
+func (dm *DockerManager) verifyNonRoot(container *api.Container) error {
+	if securitycontext.HasRootRunAsUser(container) {
+		return fmt.Errorf("Container's runAsUser breaks non-root policy.")
+	}
+
+	imgRoot, err := dm.isImageRoot(container.Image)
+	if err != nil {
+		return err
+	}
+	if !securitycontext.HasRunAsUser(container) && imgRoot {
+		return fmt.Errorf("Container has no runAsUser and image will run as root.")
+	}
+
+	return nil
+}
+
+// isImageRoot returns true if the user directive is not set on the image, the user is set to 0
+// or the user is set to root.  If there is an error inspecting the image this method will return
+// false and return the error.
+func (dm *DockerManager) isImageRoot(image string) (bool, error) {
+	img, err := dm.client.InspectImage(image)
```

Comment:
We inspect the image in the IsImagePresent method and in the GetPodStatus method.  They both use the client this way.